python/ray/autoscaler/_private/kuberay/node_provider.py

```diff
@@ -249,17 +226,30 @@ def create_node(
             raycluster = self._get(url)
             group_name = tags["ray-user-node-type"]
             group_index, group_spec = self._get_worker_group(raycluster, group_name)
-            path = f"/spec/workerGroupSpecs/{group_index}/replicas"
+            prefix = f"/spec/workerGroupSpecs/{group_index}"
             payload = [
-                {"op": "test", "path": path, "value": group_spec["replicas"]},
+                # This test makes sure that the previous request to Kuberay has been completed.
+                {
+                    "op": "test",
+                    "path": prefix + "/scaleStrategy",
+                    "value": {"workersToDelete": []},
+                },
+                # We only submit the request for additional replicas if the above test passes.
                 {
                     "op": "replace",
-                    "path": path,
+                    "path": prefix + "/replicas",
                     "value": group_spec["replicas"] + count,
                 },
+                # We insert a dummy node to delete so future requests from create_node and
+                # terminate_nodes can ensure that this request has been completed (by checking that
+                # workersToDelete is empty).
+                {
+                    "op": "replace",
+                    "path": prefix + "/scaleStrategy",
+                    "value": {"workersToDelete": [NOT_A_NODE_NAME]},
```

Comment:
This is a hack in the usual sense: 
We're achieving a goal (awaiting an ack from the operator)
by means of a semantically unrelated action (writing and reading to workersToDelete).

Basically, we're using workersToDelete to effect a task queue of capacity 1 for the autoscaler's scale requests.
 
Can we add a todo that we do not intend to do things this way forever?

The correct way is to get an ack by reading something from the CR's status field -- this will require changes to CRD.
cc @Jeffwan

Reply:
(This PR's method is of course a less error-prone hack than `_wait_for_pods`.)